### PR TITLE
Colorspace: adding abstraction of publishing related functions

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
The functionality of Colorspace has been abstracted for greater usability.

## Additional info
- publishing functions from mixin were moved to colorspace.py module

## Testing notes:
1. all should work as it was before

## Dependencies
Changes from this PR were initially implemented in [this pull request](https://github.com/ynput/OpenPype/pull/5451). However, due to its significant size, we made the decision to divide it into smaller portions.

### CODE REVIEW

File: openpype/pipeline/colorspace.py [hunk 1]
```diff
@@ -601,3 +606,133 @@ def _get_imageio_settings(project_settings, host_name):
...
+def get_colorspace_settings_from_publish_context(context_data):
+    """Returns solved settings for the host context.
+
+    Args:
+        context_data (publish.Context.data): publishing context data
+
+    Returns:
+        tuple | bool: config, file rules or None
+    """
+    if "imageioSettings" in context_data:
```
Comment (Minkiu) [inline openpype/pipeline/colorspace.py:620]: Maybe do `if "imageioSettings" in context_data and context_data["imageioSettings"]` to see that the key is not empty?

File: openpype/pipeline/colorspace.py [hunk 2]
```diff
@@ -601,3 +606,133 @@ def _get_imageio_settings(project_settings, host_name):
...
+    )
+
+    # in case host color management is not enabled
+    if not config_data:
+        return None
+
+    file_rules = get_imageio_file_rules(
+        project_name, host_name,
+        project_settings=project_settings_
+    )
```
Comment (iLLiCiTiT) [inline openpype/pipeline/colorspace.py:641]: I would suggest 2 changes:
1. Always return tuple of 2 items so you can do everywhere
    `config_data, file_rules = get_colorspace_settings_from_publish_context(...)`
2. Cache invalid state too, so it's not recalculated all the time

```suggestion
    file_rules = None
    if config_data:
        file_rules = get_imageio_file_rules(
            project_name, host_name,
            project_settings=project_settings_
        )
```

File: openpype/pipeline/colorspace.py [hunk 3]
```diff
@@ -601,3 +606,133 @@ def _get_imageio_settings(project_settings, host_name):
...
+    # caching settings for future instance processing
+    context_data["imageioSettings"] = (config_data, file_rules)
+
+    return config_data, file_rules
+
+
+def set_colorspace_data_to_representation(
+    representation, context_data,
+    colorspace=None,
+    colorspace_settings=None,
```
[the remark this answers is not in the document]
Comment (jakubjezek001) [reply openpype/pipeline/colorspace.py:652]: It should definitely benefit if a new collector for `config_data` and `file_rules` were placed higher in the collecting order. This way, they could be easily accessed from [[context.data](http://context.data/)](http://context.data).

**Review by Minkiu — APPROVED** [review]

File: openpype/pipeline/colorspace.py [hunk 4]
```diff
@@ -13,12 +13,17 @@
...
 from openpype.pipeline import Anatomy
+from openpype.lib.transcoding import VIDEO_EXTENSIONS, IMAGE_EXTENSIONS
+
 
 log = Logger.get_logger(__name__)
 
 
-class CashedData:
+class CachedData:
     remapping = None
```
Comment (BigRoy) [inline openpype/pipeline/colorspace.py:23]:
```suggestion
    remapping = {}
```

Where are we setting the `remapping` attribute to `dict`?
We can do `CachedData.remapping.setdefault`, which would then be: `None.setdefault(key, {})`

Comment (BigRoy) [reply openpype/pipeline/colorspace.py:23]: I'm pretty sure e.g. `get_remapped_colorspace_to_native` won't work currently.

Comment (jakubjezek001) [reply openpype/pipeline/colorspace.py:23]: Great catch, thanks!

File: openpype/pipeline/colorspace.py [hunk 5]
```diff
@@ -601,3 +606,118 @@ def _get_imageio_settings(project_settings, host_name):
...
+    """Returns solved settings for the host context.
+
+    Args:
+        context_data (publish.Context.data): publishing context data
+
+    Returns:
+        tuple | bool: config, file rules or None
+    """
+    if "imageioSettings" in context_data and context_data["imageioSettings"]:
+        return context_data["imageioSettings"]
```
Comment (BigRoy) [inline openpype/pipeline/colorspace.py:621]: This check basically equals `if context_data.get("imageioSettings")`

**Pull request merged**